src/app/wallets/update-on-chain-receipt.ts

```diff
@@ -219,23 +240,31 @@ const processTxForWallet = async (
             return result
           }
 
-          const recipientAccount = await AccountsRepository().findById(wallet.accountId)
-          if (recipientAccount instanceof Error) return recipientAccount
-
           const recipientUser = await UsersRepository().findById(
             recipientAccount.kratosUserId,
           )
           if (recipientUser instanceof Error) return recipientUser
 
           const displayPaymentAmount = {
-            amount: Number((amountDisplayCurrency / 100).toFixed(2)),
+            amount: Number(
+              minorToMajorUnit({
+                amount: creditAccountAdditionalMetadata.displayAmount,
+                displayMajorExponent: MajorExponent.STANDARD,
+              }),
+            ),
             currency: displayCurrency,
           }
 
+          const paymentAmount = paymentAmountFromNumber({
+            amount: sats,
+            currency: wallet.currency,
+          })
```

Comment:
I know this is only a refactor but.. shouldn't get conversion for USD wallets?

Reply:
Oh good catch, this whole flow needs to be fixed to handle USD onchain receipts